animation-proposal.md

```diff
@@ -0,0 +1,338 @@
+ * @param options.iterationCount How many times to run the interpolation. Can be a number, or "infinite". Default: 1.
+ * @param options.direction The direction the interpolation should be played.
+ */
+public interpolate(
+	selector: string,
+	value: InterpolationValueType,
+	interval: number | null,
+	options?: {
+		curve: InterpolationCurve = "linear",
+		basis: InterpolationBasis = "absolute",
+		iterationCount: InterpolationIterationCount = 1,
+		direction: InterpolationDirection = "forward"
+	}): Promise<void>;
+```
+
+
+Several new types were introduced in the definition of `interpolate` above. Expanding them:
+
+```ts
+/**
+ * A union of the different types that can be passed to the
+ * `interpolate` method's `value` parameter.
+ */
+export type InterpolationValueType
+	/** A string type indicates the value is a field selector. */
+	= string
+	/** The rest represent concrete final values. */
+	| Quaternion
+	| Vector3
+	| Vector2
+	| Color
+	| number
+	;
+
+/**
+ * Describes an interpolation curve, predefined or custom.
+ * Predefined/common curves can be specified by name, or custom curve parameters can be given directly.
+ */
+export type InterpolationCurve
+	/**
+	 * Moves from beginning to end at a constant rate.
+	 * Equivalent to: [0.0, 0.0, 1.0 ,1.0]
+	 */
+	= "linear"
+
+	/**
+	 * The animation starts slowly, accelerates sharply, and then slows
+	 * gradually towards the end. It is similar to `ease-in-out`, though
+	 * it accelerates more sharply at the beginning.
+	 * Equivalent to: [0.25, 0.1, 0.25, 1.0]
+	 */
+	| "ease"
+
+	/**
+	 * The animation starts slowly, and then progressively speeds up until
+	 * the end, at which point it stops abruptly.
+	 * Equivalent to: [0.42, 0.0, 1.0, 1.0]
+	 */
+	| "ease-in"
+
+	/**
+	 * The animation starts slowly, speeds up, and then slows down towards
+	 * the end. At the beginning, it behaves like ease-in; at the end, it
+	 * behaves like ease-out.
+	 * Equivalent to: [0.42, 0.0, 0.58, 1.0]
+	 */
+	| "ease-in-out"
+
+	/**
+	 * The animation starts abruptly, and then progressively slows down
+	 * towards the end.
+	 * Equivalent to: [0.0, 0.0, 0.58, 1.0]
+	 */
+	| "ease-out"
+
+	/**
+	 * Define a custom curve by providing P1 and P2 of a cubic bezier
+	 * curve. To be valid, each point's abscissa must be in [0, 1].
+	 * The ordinate may be outside this range; if so then you will
+	 * get a bounce effect.
+	 */
+	| number[];
+
+/**
+ * Describes how the `value` parameter to `interpolate` should be interpreted.
+ */
+export type InterpolationBasis
+	/** The value is an absolute value. The function operates as "Interpolate to this value". */
+	= "absolute"
+
+	/** The value is relative to the starting value. The function operates as "Interpolate by this much". */
+	| "relative"
```

Comment:
Does this apply to quaternions?

Reply:
Quaternions are a weird one. I want to support relative rotations. I wonder if we need to support passing a set of Euler angles as the `to` value.